spec/measure_repository_service_test_kit/measure_data_requirements_spec.rb

```diff
@@ -0,0 +1,394 @@
+      stub_request(:get, "#{url}/Measure/#{measure_id}")
+        .to_return(status: 200, body: test_measure.to_json)
+
+      stub_request(
+        :post,
+        "#{url}/Measure/#{measure_id}/$data-requirements"
+      )
+        .to_return(status: 200, body: test_library_response.to_json)
+
+      result = run(test, url:,
+                         measure_id:)
+
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if request returns 400' do
+      stub_request(
+        :post,
+        "#{url}/Measure/#{measure_id}/$data-requirements"
+      )
+        .to_return(status: 400, body: test_library_response.to_json)
+
+      result = run(test, url:,
+                         measure_id:)
+
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if request returns non-Library resource' do
+      stub_request(
+        :post,
+        "#{url}/Measure/#{measure_id}/$data-requirements"
+      )
+        .to_return(status: 200, body: test_measure.to_json)
+
+      result = run(test, url:,
+                         measure_id:)
+
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if returned Library is not of type module-definition' do
+      stub_request(
+        :post,
+        "#{url}/Measure/#{measure_id}/$data-requirements"
+      )
+        .to_return(status: 200, body: test_library_wrong_type.to_json)
+
+      result = run(test, url:,
+                         measure_id:)
+
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server successfully returns 200 and Library on $data-requirements with url' do
+    let(:test) { group.tests[1] }
+    let(:measure_url) { 'measure_url' }
+    let(:measure_version) { 'measure_version' }
+
+    it 'passes if 200 response returned with Library body given url and version' do
+      stub_request(
+        :post,
+        "#{url}/Measure/$data-requirements"
+      )
+        .to_return(status: 200, body: test_library_response.to_json)
+
+      result = run(test, url:,
+                         measure_url:,
+                         measure_version:)
+
+      expect(result.result).to eq('pass')
+    end
+
+    it 'passes if 200 response returned with Library body given just url' do
+      stub_request(
+        :post,
+        "#{url}/Measure/$data-requirements"
+      )
+        .to_return(status: 200, body: test_library_response.to_json)
+
+      result = run(test, url:,
+                         measure_url:)
+
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if 200 is not retuned' do
+      stub_request(
+        :post,
+        "#{url}/Measure/$data-requirements"
+      )
+        .to_return(status: 400, body: test_library_response.to_json)
+
+      result = run(test, url:,
+                         measure_url:,
+                         measure_version:)
+
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if Library is not retuned' do
```

Comment:
```suggestion
    it 'fails if Library is not returned' do
```
vscode spell checker didn't have your back this time